sound/soc/intel/common/soc-acpi-intel-tgl-match.c

```diff
@@ -383,6 +397,18 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_tgl_sdw_machines[] = {
 		.drv_name = "sof_sdw",
 		.sof_tplg_filename = "sof-tgl-rt711-rt1316-rt714.tplg",
 	},
+	{
+		/*
+		 * This needs to be placed before tgl_rvp since it's a
+		 * subset with a single amplifier. Inverting the
+		 * entries will result in a confusion where the same rt1308 is
+		 * counted twice.
+		 */
```

Comment:
We will not need this statement when https://github.com/thesofproject/linux/pull/2721 is merged

Reply:
On second thought, I think we can not merge this PR before #2721. I believe that TGL RVP will match the tgl_hp links since  we can find rt1308 on TGL RVP.

Reply:
no, there should be no match, because we don't have all the devices reported by hardware (i.e. the second rt1308 is not in the tgl-hp descriptor)

Reply:
> no, there should be no match, because we don't have all the devices reported by hardware (i.e. the second rt1308 is not in the tgl-hp descriptor)

But, link_slaves_found() only check the devices in the descriptor, it doesn't care about other devices which are not listed in the  descriptor. i.e. It will check the 1st rt1308 and return true since it did find rt1308. It will never check the 2nd rt1308 when it went through the tgl-hp descriptor.
I just triggered tests with and without this PR, and I can see below with this PR
```
kernel: [    4.551250] sof-audio-pci 0000:00:1f.3: found 711 at link 0
kernel: [    4.551254] sof-audio-pci 0000:00:1f.3: found 1308 at link 1
kernel: [    4.551256] sof-audio-pci 0000:00:1f.3: SoundWire machine driver sof_sdw topology sof-tgl-rt711-rt1308.tplg
``` 
which matched tgl_hp[].

And below without this PR
```
kernel: [    4.585386] sof-audio-pci 0000:00:1f.3: found 711 at link 0
kernel: [    4.585389] sof-audio-pci 0000:00:1f.3: found 1308 at link 1
kernel: [    4.585390] sof-audio-pci 0000:00:1f.3: found 1308 at link 1
kernel: [    4.585390] sof-audio-pci 0000:00:1f.3: SoundWire machine driver sof_sdw topology sof-tgl-rt711-rt1308.tplg
```
which matched tgl_rvp[].

Reply:
you're right @bardliao, we need to fix this with PR #2721 
Thanks for checking and pushing back on my flawed assumption, much appreciated.

Reply:
fixed now.